src/costmap/src/Costmap.cpp

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "costmap/Costmap.hpp"
+#include <tf2/LinearMath/Quaternion.h>
+
+using std::vector;
+
+// TODO(orduno): Port ROS1 Costmap package
+Costmap::Costmap(rclcpp::Node* node) : node_(node)
+{
+  RCLCPP_INFO(node_->get_logger(), "Costmap::Costmap");
+}
+
+void
+Costmap::getCostmap(const nav2_msgs::msg::Costmap& specifications, nav2_msgs::msg::Costmap& costmap)
+{
+  RCLCPP_INFO(node_->get_logger(), "Costmap::getCostmap");
+
+  // TODO(orduno): faking out a costmap for now
+  (void)specifications;
+
+  costmap.header.stamp = node_->now();
+  costmap.header.frame_id = "/map";
+
+  // Fill map metadata
+  costmap.info.map_load_time = node_->now();
+
+  // Some arbitrary numbers
+  costmap.info.resolution = 1;  // m/cell
+  costmap.info.width = 10;  // cells
+  costmap.info.height = 10;  // cells
+
+  // The origin of the map [m, m, rad]. This is the real-world pose of the cell (0,0) in the map.
+  // Origin is lower-left pixel?
+  costmap.info.origin.position.x = 0.0; // 2D pose of the lower-left pixel in the map
+  costmap.info.origin.position.y = 0.0;
+  costmap.info.origin.position.z = 0.0; // ignored
+
+  // Define map rotation
+  // Provided as yaw with counterclockwise rotation, with yaw = 0 meaning no rotation
+
+  tf2::Quaternion quaternion;
+  quaternion.setRPY(0.0, 0.0, 0.0); // set roll, pitch, yaw
+  costmap.info.origin.orientation.x = quaternion.x();
+  costmap.info.origin.orientation.y = quaternion.y();
+  costmap.info.origin.orientation.z = quaternion.z();
+  costmap.info.origin.orientation.w = quaternion.w();
+
+  costmap.data.resize(costmap.info.width, costmap.info.height);
+
+  // Fill with some fake data for testing
+  getTestData(costmap.info.width, costmap.info.height, costmap.data);
+}
+
+void
+Costmap::getTestData(const int width, const int height, vector<uint8_t>& data)
+{
+  // TODO(orduno): fixed size for now
+  (void)width;
+  (void)height;
+
+  // TODO(orduno): instead of hardcoding, define a function
+
+  const uint8_t n = 255; // no information
```

Comment:
Each of these special values should have a symbol. It could be from an enumeration (with explicit values) or a macro. That way, the code wouldn't need comments and would be clearer.

Reply:
Changed to enum.

Reply:
After further discussion, it was changed to static const within Costmap class